### PR TITLE
[all] Remove SOFA_FLOAT (all steps)

### PR DESCRIPTION
**Pull request by damienmarchal**
Follow issue #495 , implements the removal.

Changelog:
- Remove SOFA_FLOAT specific code everywhere.
- Remove all template instantiation based on double type.
- All types now have strict version eg: Vec3f, Vec3d and well as an un-defined precision one Vec3. This is implemented without #ifdef 
- All the component registred to the ObjectFactory are now only using the un-defined precision version of their template. 
- Add a mechanism to TemplateAlias.cpp so we can print a warning 
- In sofa scenes, both xxtypexxf and xxtypexxd  are translated into xxtypexx using the TemplateAlias mechanisme. If the conversion involve a precision change a warning is emitted. 
- Clean a sources by removing commented code.
- Deprecate the use of #include <sofa/defaulttype/Vec3Types.h> and use #include <sofa/defaulttype/VecTypes.h> (because using two files pointing to the same thing is only adding noise).

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by damienmarchal** [comment]
[ci-build][[with-scene-tests]

**Comment by damienmarchal** [comment]
[ci-build][[with-scene-tests]

**Comment by epernod** [comment]
Nice work, do you need help to test?

**Comment by damienmarchal** [comment]
Hi Erik,

Thank you for the offer,
If you have time you could look at the failing scenes to see what is wrong and how it would be best to validate that. Maybe testing this branch with your plugins or scenes would also be very useful.

DM.

File: SofaKernel/modules/SofaBaseLinearSolver/SofaBaseLinearSolver_test/Matrix_test.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@
 /** Sparse matrix test suite.
  *
  * The same suite is instanciated using different parameters: entry types
- * (float/double) and BlockMN size in CompressedRowSparse.
+ * (float/Real) and BlockMN size in CompressedRowSparse.
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseLinearSolver/SofaBaseLinearSolver_test/Matrix_test.cpp:26]: (Real)

**Comment by damienmarchal** [comment]
I'm very happy to say that it compiles and passes the tests. 
bye bye the sofa float ugliness. 
@guparan, @hugtalbot, @ChristianDuriez, @bcarrez 

So now how do we merge ?

**Comment by damienmarchal** [comment]
[ci-builds][with-scene-tests]

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests]

**Comment by damienmarchal** [comment]
To me, here are the important files to review if you agree with the changes made:
In SofaKernel:
- Vec.h
- VecTypes.h
- Mat.h
- Quat.h
- RigidTypes.h
- TemplateAlias.cpp 
- TemplateAlias.h

In Flexible
- StrainTypes
- DeformationGradientTypes

**Comment by damienmarchal** [comment]
I'm really proud of this PR that touch 950 files and does not break too much scenes.

**Pull request merged**